### DISCUSSION TITLE
Persistent view with removed buttons is bringing those buttons back after a restart and interaction

### DISCUSSION THREAD

**Discussion by wcurti** [Pycord-Development/pycord#2225] [Q&A] [answered]
Not sure if this is working as intended or not. 

1. My View is persistent, I can test this with the .is_persistent() method.
2. The View contains several buttons that are defined in its class and include custom_ids.
3. Not all of those buttons are needed in every message. When they aren't, I initialize a new View, .remove_item all the buttons that are not necessary and then send the message with the edited view.
4. If, however, the bot disconnects and reconnects, if a user clicks on a button that remains in that edited view, all of the removed buttons return.

This is my view and a test button that prints info about the view:
```
class buttonView(discord.ui.View):
    def __init__(self):
        super().__init__(timeout=None)

    @discord.ui.button(label="1", custom_id="R1")
    async def r1_callback(self, button, interaction):
            view = self
            print(str(view.is_persistent()))
            buttons = view.children

            for old_button in buttons:
                print(old_button.custom_id)

            await interaction.response.send_message("Printing details.")
```

It prints True and then a list of every button, regardless of whether or not they're removed. That means they're being recreated before I even get the view from self. I also don't need to edit the message for them to pop back up, they just do it. All of the rest of my button functionality works as intended, even after a restart.

Any ideas?

**Comment by Lulalaby** [comment, marked as answer]
The correct place to ask is https://discord.gg/pycord.

If u believe it's a bug by us, open an issue.
